ibis/expr/types/core.py

```diff
@@ -76,35 +76,41 @@ def __coerce__(cls, value):
             raise CoercionError("Unable to coerce value to an expression")
 
     def __repr__(self) -> str:
-        if not opts.interactive:
-            return self._repr()
-
-        from ibis.expr.types.pretty import simple_console
-
-        with simple_console.capture() as capture:
-            try:
-                simple_console.print(self)
-            except TranslationError as e:
-                lines = [
-                    "Translation to backend failed",
-                    f"Error message: {e!r}",
-                    "Expression repr follows:",
-                    self._repr(),
-                ]
-                return "\n".join(lines)
-        return capture.get().rstrip()
+        from ibis.expr.format import pretty
+
+        if opts.repr.show_variables:
+            import inspect
+
+            # get the variables available in the caller's frame
+            frame = inspect.currentframe().f_back
```

Comment:
Good times.

Reply:
While this may not be supported in certain python implementations, I don't think that we need to be foolproof here since this functionality is entirely opt-in.

Reply:
Seems like this isn't sufficient for an IPython session:

```
In [1]: import ibis

In [2]: t = ibis.table({"a": "int", "b": "string"})

In [3]: tt = t.mutate(x=1000000000000000000)

In [4]: ttt = tt.mutate(y="s")

In [5]: tt
Out[5]:
r0 := UnboundTable: unbound_table_0
  a int64
  b string

Project[r0]
  a: r0.a
  b: r0.b
  x: 1000000000000000000

In [6]: ttt
Out[6]:
r0 := UnboundTable: unbound_table_0
  a int64
  b string

r1 := Project[r0]
  a: r0.a
  b: r0.b
  x: 1000000000000000000

Project[r1]
  a: r1.a
  b: r1.b
  x: r1.x
  y: 's'
```